app/models/usage_update.rb

```diff
@@ -0,0 +1,213 @@
+
+  def self.import_by_month(options={})
+    from_date = (options[:from_date].present? ? Date.parse(options[:from_date]) : Date.current).beginning_of_month
+    until_date = (options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current).end_of_month
+
+    # get first day of every month between from_date and until_date
+    (from_date..until_date).select {|d| d.day == 1}.each do |m|
+      UsageUpdateImportByMonthJob.perform_later(from_date: m.strftime("%F"), until_date: m.end_of_month.strftime("%F"))
+    end
+
+    "Queued import for Report updated from #{from_date.strftime("%F")} until #{until_date.strftime("%F")}."
+  end
+
+  def self.import(options={})
+    # from_date = options[:from_date].present? ? Date.parse(options[:from_date]) : Date.current - 1.day
+    # until_date = options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current
+
+    usage_update = UsageUpdate.new
+    # usage_update.queue_jobs(usage_update.unfreeze(from_date: from_date.strftime("%F"), until_date: until_date.strftime("%F")))
+    usage_update.queue_jobs 
+  end
+
+  def source_id
+    "usage_update"
+  end
+
+
+  def process_data options={} 
+    messages = get_query_url options
+    messages.each do |message|
+      body = JSON.parse(message.body)
+      report_id = body["report_id"]
+      UsageUpdateParseJob.perform_later(report_id, options)
+      delete_message message
+    end if messages.respond_to?("each")
+    messages.length
+  end
+
+  def get_query_url options={}
+    queue_url = sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    resp = sqs.receive_message(queue_url: queue_url, max_number_of_messages: 5, wait_time_seconds: 1)
+    resp.messages
+  end
+
+  def self.get_data report_id, options={}
+    host = URI.parse(report_id).host.downcase
+    report = Maremma.get(report_id, timeout: 120, host: host)
+    report
+  end
+
+  def sqs
+    sqs = Aws::SQS::Client.new(region: ENV["AWS_REGION"])
+    puts sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    sqs
+  end
+
+
+  def get_total options={}
+    queue_url = sqs.get_queue_url(queue_name: "#{Rails.env}_usage" ).queue_url
+    req = sqs.get_queue_attributes(
+      {
+        queue_url: queue_url, attribute_names: 
+          [
+            'ApproximateNumberOfMessages', 
+            'ApproximateNumberOfMessagesNotVisible'
+          ]
+      }
+    )
+
+    msgs_available = req.attributes['ApproximateNumberOfMessages']
+    msgs_available.to_i
+  end
+
+  def queue_jobs(options={})
+
+    total = get_total(options)
+    
+    if total < 1
+      text = "No works found for in the Usage Reports Queue."
+    end
+
+    num_messages = total
+    while num_messages > 0 
+        queued = process_data(options)
+        num_messages -= queued
+        puts num_messages
+        puts queued
+    end
+    text = "#{queued} reports queued out of #{total} for Usage Reports Queue"
+
+    Rails.logger.info text
```

Comment:
Rails.logger only works in controllers. Please define a logger that logs to stdout.